app/screens/sso/sso_with_redirect_url.tsx

```diff
@@ -0,0 +1,178 @@
+        });
+        const url = parsedUrl.toString();
+        Linking.canOpenURL(url).then(() => {
+            Linking.openURL(url);
+        }).catch(() => {
+            setError(
+                intl.formatMessage({
+                    id: 'mobile.oauth.failed_to_open_link',
+                    defaultMessage: 'Failed to open the link',
+                }),
+            );
+        });
+    };
+
+    const onURLChange = ({url}: { url: string }) => {
+        if (url && url.startsWith(redirectUrl)) {
+            Store?.redux?.dispatch(setDeepLinkURL(''));
+            const parsedUrl = urlParse(url, true);
+            if (parsedUrl.query && parsedUrl.query.MMCSRF && parsedUrl.query.MMAUTHTOKEN) {
+                onCSRFToken(parsedUrl.query.MMCSRF);
+                onMMToken(parsedUrl.query.MMAUTHTOKEN);
+            } else {
+                setError(
+                    intl.formatMessage({
+                        id: 'mobile.oauth.failed_to_login',
+                        defaultMessage: 'Failed to login',
+                    }),
+                );
+            }
+        }
+    };
+
+    React.useEffect(() => {
+        Linking.addEventListener('url', onURLChange);
+        init(false);
+        return () => {
+            Linking.removeEventListener('url', onURLChange);
+        };
+    }, []);
+
+    return (
+        <SafeAreaView
+            style={style.container}
+            testID='sso.redirect_url'
+        >
+            {loginError || error ? (
+                <View style={style.errorContainer}>
+                    <View style={style.errorTextContainer}>
+                        <Text style={style.errorText}>{loginError || error}{'.'}</Text>
+                    </View>
+                    <TouchableOpacity onPress={() => init()}>
+                        <FormattedText
+                            id='mobile.oauth.try_again'
+                            defaultMessage='Try again'
+                            style={style.button}
+                        />
+                    </TouchableOpacity>
+                </View>
+            ) : (
+                <View style={style.infoContainer}>
+                    <FormattedText
+                        id='mobile.oauth.switch_to_browser'
+                        defaultMessage='Switch to browser and complete the login process'
+                        style={style.infoText}
+                    />
+                    <FormattedText
+                        id='mobile.oauth.or'
+                        defaultMessage='or'
+                        style={style.infoText}
+                    />
+                    <TouchableOpacity onPress={() => init()}>
+                        <FormattedText
+                            id='mobile.oauth.restart_login'
+                            defaultMessage='Restart login'
+                            style={style.button}
+                        />
+                    </TouchableOpacity>
```

Comment:
this can definitely break localization, please give this blog a read https://mattermost.com/blog/avoiding-common-internationalization-mistakes/

Reply:
Thanks! I have read the blog but couldn't understand the mistake I made here.
These 3 lines are independent lines as shown in the image. 
<img width="312" alt="Screenshot 2021-01-05 at 7 42 24 PM" src="https://user-images.githubusercontent.com/3680799/103656107-3189d900-4f8e-11eb-9c2c-7776cf504366.png">

Reply:
Perhaps add the `or` with a new line instead of a separate string